spec.md

```diff
@@ -91,13 +91,37 @@ perform a scheduled reboot.
 #### Event
 
 An "event" is a data record expressing an occurrence and its context. Events are
-routed from an event producer (the source) to interested event consumers. The
-routing can be performed based on information contained in the event, but an
-event will not identify a specific routing destination. Events will contain two
-types of information: the [Data](#data) representing the Occurrence and
+routed from an event producer to interested event consumers. The routing can be
+performed based on information contained in the event, but an event will not
+identify a specific routing destination. Events will contain two types of
+information: the [Data](#data) representing the Occurrence and
 [Context](#context) metadata providing contextual information about the
 Occurrence.
 
+#### Producer
+
+The "producer" is a specific instance, process or device that creates the data
+structure describing the event and its context.
+
+#### Source
+
+The "source" is the logical system or service where the occurrence happened. In
+a distributed system it might consist of multiple [Producers](#producer). If a
+source is not aware of CloudEvents, a producer creates the CloudEvent on behalf
+of the source.
+
+#### Consumer
+
+A "consumer" receives the event and acts upon it. It uses the content and data
```

Comment:
not sure what you mean by "content and data" - do you mean "context attribute and data attribute" ?

Reply:
Or if you want to just say "Context" instead of "Context attributes" then perhaps make it a "C" instead of "c" and make it an href to the "Context" definition. It's not clear when you mean the CE Context entity vs something else.

Reply:
Yes, that's a general issue about the terminology. If you look at the definition of _event_, it is already CloudEvent specific, explaining context and data. Does that mean that whenever I mention _event_, I imply that it is a CloudEvent?